src/contracts/SafeEntrypoint.sol

```diff
@@ -62,68 +52,49 @@ contract SafeEntrypoint is SafeManageable, ISafeEntrypoint {
     allowedActions[_actionContract] = false;
   }
 
+  /// @inheritdoc ISafeEntrypoint
+  function unqueueActionContract(address _actionContract) external isAuthorized {
+    queuedActions[_actionContract] = false;
+  }
+
   // ~~~ ACTIONS METHODS ~~~
 
   /// @inheritdoc ISafeEntrypoint
   function queueApprovedAction(address _actionContract) external isAuthorized returns (bytes32 _actionHash) {
-    if (!allowedActions[_actionContract]) revert NotAllowed();
-
-    IActions.Action[] memory actions = IActions(_actionContract).getActions();
-    _actionHash = keccak256(abi.encode(actions, _actionNonce++));
-
-    uint256 _executableAt = block.timestamp + 1 hours;
-    actionExecutableAt[_actionHash] = _executableAt;
-    actionData[_actionHash] = abi.encode(actions);
-
-    // NOTE: event picked up by off-chain monitoring service
-    emit ApprovedActionQueued(_actionHash, _executableAt);
+    address[] memory _actionContracts = new address[](1);
+    _actionContracts[0] = _actionContract;
+    return queueApprovedActions(_actionContracts);
   }
 
-  function getBatchedActions(address[] memory actionContracts) public /* view */ returns (IActions.Action[] memory) {
-    uint256 _totalLength;
-    IActions.Action[][] memory _cachedActions = new IActions.Action[][](actionContracts.length);
-
-    // First pass: call getActions once per contract and cache the results
-    for (uint256 _i; _i < actionContracts.length; ++_i) {
-      address _actionContract = actionContracts[_i];
-      if (!allowedActions[_actionContract]) revert NotAllowed();
-      if (queuedActions[_actionContract]) revert ActionAlreadyQueued();
-      IActions.Action[] memory actions = IActions(_actionContract).getActions();
-
-      queuedActions[_actionContract] = true;
+  /// @inheritdoc ISafeEntrypoint
+  function queueApprovedActions(address[] memory _actionContracts) public isAuthorized returns (bytes32 _actionHash) {
```

Comment:
we usually avoid having `public` methods, idk if there are strong opinions about, but ends up being clearer to test and manipulate

Reply:
Switched to external, and removed the `queueApprovedAction` method, now both methods are plural, when queueing just 1 approved action contract we can just pass an array of length 1.